Sources/FutureComposition.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  FutureComposition.swift
+//  Deferred
+//
+//  Created by Zachary Waldowski on 4/2/16.
+//  Copyright © 2014-2016 Big Nerd Ranch. Licensed under MIT.
+//
+
+extension FutureType {
+    /// Composes this future with another.
```

Comment:
Yeah, I think these can do away with the doc tags (except for `seealso`) and just be "Creates a future that becomes…"

[See also](https://swift.org/documentation/api-design-guidelines/):

> Describe what a function or method does and what it returns, omitting null effects and Void returns:
> 
> ```swift
> /// Inserts `newHead` at the beginning of `self`.
> mutating func prepend(newHead: Int)
> 
> /// Returns a `List` containing `head` followed by the elements
> /// of `self`.
> func prepending(head: Element) -> List
> 
> /// Removes and returns the first element of `self` if non-empty;
> /// returns `nil` otherwise.
> mutating func popFirst() -> Element?
> ```